tests/lookup/tests.py

```diff
@@ -956,3 +957,15 @@ def test_nested_outerref_lhs(self):
             ),
         )
         self.assertEqual(qs.get(has_author_alias_match=True), tag)
+
+    def test_exact_selected_field_rhs_subquery(self):
+        """Ensures GROUP BY in query persists when used as subquery."""
+        _ = Author.objects.create(name='one')
+        author_2 = Author.objects.create(name='two')
+        max_ids = Author.objects.values('alias').annotate(m=Max('id')).values('m').order_by()
+        # Include empty .order_by() to disregard ordering set in Author's Meta class.
+
+        self.assertEqual('SELECT MAX("lookup_author"."id") AS "m" FROM "lookup_author" GROUP BY "lookup_author"."alias"', str(max_ids.query))
```

Comment:
Don't assert against the exact SQL since per-backend dialect will have a different syntax (e.g. wrt to identifier quoting).

```suggestion
```

Asserting against the resultset should be enough.

Reply:
@charettes Thanks, yeah I added that in bc tests passed locally, but failed on MariaDB in CI, so I added that line assuming the assertion would fail and I'd get to see what query is actually getting output.